CMakePresets.json

```diff
@@ -0,0 +1,92 @@
+{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "stm32f302-preset",
+      "displayName": "STM32F302x8 Build",
+      "description": "Configures the build for an STM32F302x8 target using the ARM GCC toolchain and Ninja.",
+      "generator": "Ninja",
+      "binaryDir": "${sourceDir}/build/${presetName}",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "EVT_CORE_LOG_ENABLE": "1",
+        "TARGET_DEV": "STM32F302x8",
+        "FORCE_COLORED_OUTPUT": "1"
+      },
+      "environment": {
+        "GCC_ARM_TOOLS_PATH": "C://EVT/arm-tools/bin/"
+      }
+    },
+    {
+      "name": "stm32f334-preset",
+      "displayName": "STM32F334x8 Build",
+      "description": "Configures the build for an STM32F334x8 target using the ARM GCC toolchain and Ninja.",
+      "generator": "Ninja",
+      "binaryDir": "${sourceDir}/build/${presetName}",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "EVT_CORE_LOG_ENABLE": "1",
+        "TARGET_DEV": "STM32F334x8",
+        "FORCE_COLORED_OUTPUT": "1"
+      },
+      "environment": {
+        "GCC_ARM_TOOLS_PATH": "C://EVT/arm-tools/bin/"
```

Comment:
I am not familiar with vscode environment setup so this may not be applicable..

Is there a way to pull this from the user's environment variables? Maybe a fall back to this would be good. The default setting could lead to problems where people think they don't have the tools installed properly, but VSCode is just pointing to the wrong directory.

Reply:
Looks like the environment was just not loading right when I was testing it, but now it is pulling it in correctly.